src/devices/Dhtxx/Devices/Dht22Device.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Iot.Units;
+
+namespace Dhtxx.Devices
+{
+    public class Dht22Device : IDhtDevice
+    {
+        public double GetHumidity(byte[] readBuff)
+        {
+            return (readBuff[0] * 256 + readBuff[1]) * 0.1;
```

Comment:
readBuff[0] + readBuff[1] * 0.1;

Reply:
@ZhangGaoxing That formula is not working for me with DHT22.
The first 2 bytes (16 bits) of humidity is a 16-bit representation of 10 times the actual humidity value. So when `readBuff[0] = 00000010 (2)` and `readBuff[1] = 10011011 (155)` your formula gives the following output `2 + 155 * 0.1 = 17.5 %`. 

The right humidity should be `0000001010011010 (667) * 0.1 = 66.7 %`.

This can be achieved with the following formulas:
* `(readBuff[0] << 8 | readBuff[1]) * 0.1;`
* `(readBuff[0] * 256 + readBuff[1]) * 0.1;`

I don't know why it is working on the DHT11, do you have any suggestion?

Reply:
You are right. I just read the DHT22 datasheet and found that the formula for calculating the results is different from DHT11.

Reply:
I do have both devices now. Tested with DHT11 and DHT22 and both working fine.